src/app/multi-editor/multi-editor.component.ts

```diff
@@ -75,9 +76,14 @@ export class MultiEditorComponent implements OnInit {
     this.previewedActions = this.userActions;
     this.queryService.previewActions(this.userActions, this.lastSearchedQuery, this.currentPage, this.pageSize)
       .then((res) => {
-        this.newRecords = res['json_records'];
-        this.recordErrors = res['errors'];
-        this.previewMode = true;
+        if (res['response_text']) {
```

Comment:
this is a bad practice, both for backend and frontend.

- If there is an error, return error from backend and handle it in `catch` of promise.
- If you __have to__ return success and body that contains, `response_text` is terrible name for a key that holds the error message.